x/execution/internal/types/genesis.go

```diff
@@ -1,23 +1,38 @@
 package types
 
+import (
+	fmt "fmt"
+
+	executionpb "github.com/mesg-foundation/engine/execution"
+	"github.com/mesg-foundation/engine/ext/xvalidator"
+)
+
 // GenesisState - all instance state that must be provided at genesis
 type GenesisState struct {
-	Params Params `json:"params" yaml:"params"`
+	Params     Params                   `json:"params" yaml:"params" validate:"dive"`
+	Executions []*executionpb.Execution `json:"executions" yaml:"executions" validate:"dive"`
 }
 
 // NewGenesisState creates a new GenesisState object
-func NewGenesisState(params Params) GenesisState {
-	return GenesisState{Params: params}
+func NewGenesisState(params Params, execs []*executionpb.Execution) GenesisState {
+	return GenesisState{
+		Params:     params,
+		Executions: execs,
+	}
 }
 
 // DefaultGenesisState - default GenesisState used by Cosmos Hub
 func DefaultGenesisState() GenesisState {
 	return GenesisState{
-		Params: DefaultParams(),
+		Params:     DefaultParams(),
+		Executions: nil,
```

Comment:
Shouldn't it be an empty array to make sure we iterate over something. Otherwise `InitGenesis` will import with a nil value. Same comment for all the other modules

Reply:
It will not cause any issue within Go.
But it could cause some issue in other language or in JSON.
I will update to an empty slice to be sure it doesn't cause any issue in other languages.
Source: https://www.pixelstech.net/article/1539870875-Empty-slice-vs-nil-slice-in-GoLang